ansible/roles/cyclonedds/tasks/main.yaml

```
@@ -0,0 +1,55 @@
+- name: Create cyclonedds directory
+  ansible.builtin.file:
+    path: /opt/autoware/cyclonedds
+    state: directory
+    mode: 0755
+  become: true
+
+- name: Copy cyclonedds config
+  ansible.builtin.template:
+    src: cyclonedds.xml.jinja2
+    dest: /opt/autoware/cyclonedds/cyclonedds.xml
+    mode: 0755
+  become: true
+
+- name: Copy multicast setup service
+  ansible.builtin.copy:
+    src: multicast@.service
+    dest: /etc/systemd/system/
+    mode: 0755
+  become: true
```

I'm not familiar with how this Ansible role and service work, but `{{ item }}` isn't required here?
If not, is the file `multicast@.service` used in common for every network_interface_address in the task `Enable multicast setup service`?

```suggestion
    src: multicast@{{ item }}.service
    dest: /etc/systemd/system/
    mode: 0755
  become: true
  with_items: "{{ cyclonedds.network_interface_address }}"
```

---

@kenji-miyake Sorry for my late reply, and thanks your comments!

> is the file multicast@.service used in common for every network_interface_address in the task Enable multicast setup service?

Correct. We can either separate service file for each interface or specify with argument `%i` when enabling it.

---

Okay, thank you!